src/firebase/firebase.utils.js

```diff
@@ -0,0 +1,52 @@
+import firebase from "firebase/app";
+
+import 'firebase/firestore' ;
+import 'firebase/auth';
+
+const config = {
+        apiKey: "AIzaSyD58h2trFLWYO2MbCoPI7Jdh7ku1jcURR8",
+        authDomain: "myshop-db-1b6b9.firebaseapp.com",
+        databaseURL: "https://myshop-db-1b6b9.firebaseio.com",
+        projectId: "myshop-db-1b6b9",
+        storageBucket: "myshop-db-1b6b9.appspot.com",
+        messagingSenderId: "987153241576",
+        appId: "1:987153241576:web:3c40833ead48bc0199005d",
+        measurementId: "G-BB46WCDRZ4"
+      
+};
+firebase.initializeApp(config);
+
+export const createUserProfileDocument =async (userAuth, additionalData)=>{
+        if(!userAuth) return;
+
+        const userRef =firestore.doc(`users/${userAuth.uid}`);
+        const snapShot = await userRef.get();
+      
+        if(!snapShot.exists){
+                const {displayName , email} = userAuth;
+                const createdAt = new Date();
+
+                try{
```

Comment:
Why try catch block ? Use async catch, you're already using `await`, just use catch.